canvasobject.go

```diff
@@ -105,3 +105,9 @@ type Tabbable interface {
 type Tappable interface {
 	Tapped(*PointEvent)
 }
+
+// CacheExpirable describes any CanvasObject that can expire its cache.
+// Its methods could gain more arguments, as objects require other information to determine if the cache is expired.
+type CacheExpirable interface {
+	IsCacheExpired(Size) bool
+}
```

Comment:
Do we really need this interface? We have access to `.Size()` and the size we pass in so I think it seems unnecessary to have the interface as well.

Reply:
My understanding is that the cache texture is used by many other Canvas objects besides the `LinearGradient`/`RadialGradient`, so this interface avoids invalidating the cache for those other objects.

I haven't tested the rest of the objects that use the cache, but can go through that to understand if all of them should invalidate it on size change, or if we should maintain this as an opt-in for each Canvas object.

Reply:
It seems like there may be a simpler solution - when `Resize` of a gradient is called we can invalidate the texture with `Refresh`...

That is how image and rectangle does it

Reply:
Great feedback. I found that calling `Resize` completely fixes the issue. My previous approach was to trigger `repaint` instead, but that sometimes worked and sometimes didn't, apparently because of a race condition between marking cache as dirty and a different process actually cleaning it up?